### ISSUE TITLE
2 songs playing at the same time

### ISSUE THREAD

**Issue by DannyTalent** [theweavrs/BreadPlayer#16]
On mobile, I double clicked a song to play, but then clicked another one and now there's the first song  playing in the background, as well as the second one. I can pause/change the second one, but the first one just keeps on playing.

EDIT: Stopped playing after it finished

**Comment by theweavrs** [comment]
Try to reproduce this again to see if it is a bug or a glitch. Thank you.

**Comment by DannyTalent** [comment]
Can't seem to do it again.. I'll let you know if I find a way

**Comment by theweavrs** [comment]
I will have to check and reproduce too. Maybe it happens on PC too.

**Comment by DannyTalent** [comment]
I know what causes it!!
On mobile, double tapping the play button to play a new song causes it!!

I can't reproduce it on PC since the button jumps elsewhere when I click it

**Comment by theweavrs** [comment]
We will have to disable double tap then. Simple :laughing:

**Comment by theweavrs** [comment]
@DannyTalent Any news?

**Comment by DannyTalent** [comment]
Yeah man, sorry.. I had an exam on saturday so I was busy studying, and then I had to pick my camera up at a friend's house... Anyways, here's the video https://youtu.be/BrNmCtLO0Yg (should be up about 10 minutes after I comment this)

**Comment by theweavrs** [comment]
No problem at all, man. We are all working people with busy lives. Nothing to be sorry for, I completely understand.

**Comment by theweavrs** [comment]
@DannyTalent I think the streams aren't freeing up very nicely, that could be a problem. On the other hand, it may be device specific since it does not occur on PC or emulator. 

If you have a friend or any one who has a Windows Phone you can try and test. See if it occurs on all devices. I will search for someone willing to test this too.

Anyways, thank you so much for all the effort and bug testing. It's always a pleasure to talk with you. I have a question: Are you going through all the code of Bread Player? If so I can help you in anyway you require.

**Comment by DannyTalent** [comment]
Same dude, I'm just apologizing cause I didn't get a chance to let you know I'd be late :p anyway, sadly that probably won't be possible, but I'll keep it in mind. It may be device specific yes.. But I'm not sure. Nope, not yet, I'll probably check it this Tuesday however, since I have a lot of free time.

**Comment by theweavrs** [comment]
@DannyTalent Even so, thanks a lot. Keep hunting. I will search for some people having Windows Phone.

**Comment by DannyTalent** [comment]
Alright... I opened the project and started [reading it....](https://www.youtube.com/watch?v=Ccoj5lhLmSQ)

**Comment by Bond-009** [comment]
I was able to reproduce this issue on PC.

**Comment by Bond-009** [comment]
I just tried to reproduce it while recording it and Bread Player crashed.

**Comment by theweavrs** [comment]
@Bond-009 Can you tell me exactly how it occurred?

**Comment by Bond-009** [comment]
@theweavrs I double-click the play button of a song.

**Comment by Bond-009** [comment]
@theweavrs I recorded it: https://youtu.be/LErAQ4a366o
The video also shows that it doesn't display the cover's correctly, the song has no cover but it shows the cover from another song.

**Comment by theweavrs** [comment]
@Bond-009 It _is_ a very serious bug. I wasn't experiencing this on PC but I will try again. Also, have you updated to the latest version i.e v0.1.3-alpha? Also, I witnessed some UI problems on your screen, specifically around the Hamburger Menu. Are you using a very large screen i.e greater than 1920x1080?

Thank you so much for all the effort and yes, the covers are also not updating.

**Comment by theweavrs** [comment]
@DannyTalent @Bond-009 I will release the next alpha today with hopefully all these bugs fixed but if only I could reproduce them on my machine!

**Comment by DannyTalent** [comment]
Sweet.

**Comment by Bond-009** [comment]
@theweavrs No my main monitor is 1920x1080. This UI bug happens sometimes, not always.

**Comment by theweavrs** [comment]
I was also able to reproduce this. Its seems the previous stream isn't freed properly so that's why its causing all this.

**Comment by theweavrs** [comment]
@DannyTalent @Bond-009 I have hopefully fixed this...this time for real but I am not sure. I will upload the package and then you test.

**Comment by theweavrs** [comment]
@DannyTalent @Bond-009 New alpha is uploaded, go check it out.

**Comment by DannyTalent** [comment]
Nope. Still there, though it's behaving strangely now. First I tried reproducing it and it _kind of_ happened: It was playing the same song 2 times, however, when I paused it, it stopped both streams. When I unpaused there were still 2. Then the same behavior happened just like in the last 2 alphas..

Also, one time, I tried to double click the song and it played one in the "background" (one of the doubles) and then it started reproducing a completely different song.. Though that may have been a misread by the touchscreen

**Comment by theweavrs** [comment]
@DannyTalent As much as I hate this issue I must fix it. 😆 So, now I have uploaded the updated the packages. Try it out now. I tested it and I couldn't reproduce it. Let's see what happens.

**Comment by DannyTalent** [comment]
Alright, here's my experience:

**ON PHONE:** Pretty bad.. It happens and now the app crashes much much more
**ON PC: ** Better. It does happen but as soon as I select a new song, it doesn't anymore. Sometimes, the double click crashes the app, though

**Comment by theweavrs** [comment]
I just realized that this is not an issue and it will not occur if we load the songs normally and humanly. This only occurs when two commands are sent to load the song at almost the same time (which won't happen very often). 

I added a method to clear up channels on every song play which will avoid loading channels one after the other until we have 10 songs playing at the same time. So that is what you are actually experiencing.

About the crashes: can you specify when they happen most often?

**Comment by DannyTalent** [comment]
Well, accidental double tapping may happen, especially on mobile..
When clicking on the play button after performing a double tap, mostly

**Comment by theweavrs** [comment]
Yes, you are right. I think if we check for the channel count (the number of channels currently playing) right after a song starts playing, and then if it is more than 1 we can "delete" the second channel (which is not possible without some extensive logging).

Anyhow, I will try and work around this issue. It is not as serious on PC, I try again and again to reproduce this especially after this latest release and it doesn't appear. But on the Phone where a user can touch anything accidentally, this is serious. I will have to release a patch in an hour or two, you can test it then.

**Comment by theweavrs** [comment]
@DannyTalent Updated again! Try and see if its fixed. _I couldn't reproduce this on PC._

**Comment by DannyTalent** [comment]
Alright, it's much better now... Though it still happens on phone if you try to. On PC it's much better, it may have happened only once (I'm pretty sure)... I think this _reset_ you've done glitches out the app after a few tries, I mean I tried to play a song the buggy way, and it played one background track of that song and one "normal" stream of the first track in my library...
After a couple tries I crashed the app and then every time I double tapped, it'd crash. Playing another song fixed this

**Comment by DannyTalent** [comment]
Have you tried just disabling the button for like 0.1s after it registers a tap?

**Comment by theweavrs** [comment]
@DannyTalent yup! That's a good workaround. I will try that.

**Comment by theweavrs** [comment]
And that crash is because of Access Denied exception while saving the song info in a text file. I guess I will have to remove that.

**Comment by theweavrs** [comment]
@DannyTalent @Bond-009 @MightyK1337 Hello! I just released a new Alpha. Would you please be kind enough to give it go? Here's the link. [Alpha-1.5](https://github.com/theweavrs/BreadPlayer/releases/tag/v0.1.5-alpha)

**Comment by DannyTalent** [comment]
Alright, it either doesn't happen anymore or it's extremely more difficult to reproduce! Still however, the app crashes a lot so I can't provide much feedback :/

**Comment by theweavrs** [comment]
Crashes. Can you give me a few hints as to where the app crashes?

**Comment by DannyTalent** [comment]
I'm not being specific because it's very random... When clicking on a song, when creating a playlist, when playing a song, etc.

Anyway, can't do any more because as you can see in the new issue, I goofed it 😅

**Comment by theweavrs** [comment]
Ha! I will release a patch as a quick fix for all the crashes (exception handlers) in an hour or two. You can try it again then.

Anyhow, how do you find the new UI changes?

**Comment by DannyTalent** [comment]
I like the icons! Album view might need some more spacing on the accent-colored squares that surround the album when you select it (a bit more on the sides). Playlists menu can't be scrolled through

found a little bug, will post on the UI issue
